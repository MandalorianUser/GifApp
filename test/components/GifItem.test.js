import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe("Pruebas en GifItem",()=>{
const title = "One Punch Man";
const url = "https://www.onepunchman.com/";

    test('Debe hacer match con el snapshot',()=>{
       const {container} = render(<GifItem title={title} url={url}/>)
        expect( container ).toMatchSnapshot();
    })

    test('Debe mostar la imagen con el url y el ALT indicado', () => {
     render(<GifItem title={title} url={ url }/>)
        // expect(screen.getByRole('img').alt).toBe(title);
        // expect(screen.getByRole('img').src).toBe(url)
        const {src ,alt } = screen.getByRole('img');
        expect ( src ).toBe( url )
        expect ( alt ).toBe( title )

})

test('Debe mostrar el titulo en el componente',()=>{
    render( <GifItem title={ title} url={ url }/>)
    expect(screen.getByText( title )).toBeTruthy();
})
})