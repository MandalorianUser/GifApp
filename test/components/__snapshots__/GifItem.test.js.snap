// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man"
      src="https://www.onepunchman.com/"
    />
    <p>
      One Punch Man
    </p>
  </div>
</div>
`;
